trunner/target/armv7a9.py

```diff
@@ -0,0 +1,138 @@
+import subprocess
+import time
+from abc import abstractmethod
+from typing import Callable, Optional
+
+from trunner.config import TestContext
+from trunner.dut import Dut, SerialDut
+from trunner.harness import (
+    HarnessBuilder,
+    PloInterface,
+    PloImageLoader,
+    PloImageProperty,
+    ShellHarness,
+    Rebooter,
+    RebooterHarness,
+    FlashError,
+)
+from trunner.harness import TerminalHarness
+from trunner.host import Host
+from trunner.tools import JLinkGdbServer, Phoenixd
+from trunner.types import TestResult, TestOptions
+from .base import TargetBase, find_port
+
+
+class ARMv7A9TargetRebooter(Rebooter):
+    # TODO add text mode reboot
+
+    def _reboot_soft(self):
+        self._reboot_hard()
+
+    def _reboot_hard(self):
+        self.host.set_power(False)
+        # optimal power off time to prevent sustaining chips, e.g. flash memory, related to #540 issue
+        time.sleep(0.75)
+        self.dut.clear_buffer()
+        self.host.set_power(True)
+        time.sleep(0.05)
+
+    def _set_flash_mode(self, flash):
+        self.host.set_flash_mode(not flash)
+
+
+class ZynqGdbPloLoader(TerminalHarness, PloInterface):
+    def __init__(self, dut: Dut, script: str, cwd: Optional[str] = None):
+        TerminalHarness.__init__(self)
+        PloInterface.__init__(self, dut)
+        self.script = script
+        self.cwd = cwd
+        self.gdbserver = JLinkGdbServer("Zynq 7020")
+
+    def __call__(self):
+        """Loads plo image to RAM using gdb."""
+
+        with self.gdbserver.run():
+            try:
+                subprocess.run(
+                    ["gdb-multiarch", "plo-gdb.elf", "-x", self.script, "-batch"],
```

Comment:
The error won't be caught, when for example plo binary is missing and we won't know why the runner fail, because it will print only sth like: `expected: Waiting for input, got: 'nothing'`. It's not crucial to fix it before the merge, but maybe we could assert the following log, which seems to occur after executing a script correctly: `The current source language is "auto; currently asm".`

Reply:
This is because gdb ends with 0 error code if the binary is missing. Let's implement it in the other PR, I have to verify how gdb behaves in these cases.